experiment/ckpts/PFFx2/fromZero/config.txt
2020-05-07-19:42:59

config: <configparser.ConfigParser object at 0x7efd8f005b50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-07-19:44:08

config: <configparser.ConfigParser object at 0x7f7be816eb90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-07-19:45:24

config: <configparser.ConfigParser object at 0x7f45240b9b50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-07-19:47:58

config: <configparser.ConfigParser object at 0x7f873ad2ab50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: False
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-03:17:02

config: <configparser.ConfigParser object at 0x7f8ee4a1ab90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-03:19:57

config: <configparser.ConfigParser object at 0x7fcaa98e8b90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-03:21:52

config: <configparser.ConfigParser object at 0x7fe5be3c3b50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:29:38

config: <configparser.ConfigParser object at 0x7f2a159f5c50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: False
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:31:14

config: <configparser.ConfigParser object at 0x7f8e6c8d9c10>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:32:15

config: <configparser.ConfigParser object at 0x7f68f72dec50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:36:28

config: <configparser.ConfigParser object at 0x7f9230055cd0>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:36:53

config: <configparser.ConfigParser object at 0x7f975a46ae50>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:37:33

config: <configparser.ConfigParser object at 0x7f693803ecd0>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:38:11

config: <configparser.ConfigParser object at 0x7fd706917c90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 1
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: True
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:39:19

config: <configparser.ConfigParser object at 0x7f28551fbb90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 4
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: False
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:42:46

config: <configparser.ConfigParser object at 0x7fe3f5ebcb90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 4
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: False
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:43:21

config: <configparser.ConfigParser object at 0x7fe489f22bd0>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full
train_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/train/640_flir_hr']
test_set_paths: ['/home/ferhatcan/Image_Datasets/ir_sr_challange/test/640_flir_hr']
rgb_range: 1
batch_size: 4
scale: 2
include_noise: True
noise_sigma: 1.0
noise_mean: 0.0
include_blur: True
blur_radius: 0.2
normalize: between01
random_flips: True
channel_number: 3
n_colors: 3
hr_shape: [200, 200]
downgrade: bicubic
validation_size: 0.2
shuffle_dataset: True
model: PFF
self_ensemble: False
emb_dimension: 16
filterSize: 17
pretrained: True
learning_rate: 5e-05
decay: 200
decay_factor_gamma: 0.5
optimizer: ADAM
momentum: 0.9
betas: [0.9, 0.999]
epsilon: 1e-08
weight_decay: 0.0
gclip: 0.0
training: True
epoch_num: 100
loss: 1*PFF
skip_thr: 100000000.0
image_range: 1
log_every: 100
validate_every: 0.4
log_psnr: True
log_ssim: False
save_path: /experiment/fromZero
pre_train: load_best
only_body: False
fine_tuning: False
freeze_initial_layers: False
chop: False
save_models: False
resume: -1
test_only: False
log_test_result: True
test_single: False
test_psnr: True
test_ssim: False
test_visualize: True
test_image_save: False
load: 
save: ./ckpts/PFFx2/fromZero
reset: False
data_test: 

2020-05-08-09:45:07

config: <configparser.ConfigParser object at 0x7f0287cbbb90>
experiment_name: fromZero
device: gpu
seed: 1
n_GPUs: 1
precision: full

options.py
# It includes input arguments and options used in system
from configparser import ConfigParser


class options:
    def __init__(self, config_file_name):
        self.config = ConfigParser()
        self.config.read(config_file_name)

        self.parseDefaults()
        self.parseHarware()
        self.parseDataset()
        self.parseModel()
        self.parseOptimization()
        self.parseOptimization()
        self.parseTraining()
        self.parseTesting()
        self.parseCheckpoint()

    def parseDefaults(self):
        self.experiment_name = self.config["DEFAULT"]["experiment_name"]

    def parseHarware(self):
        self.device     = self.config["HARDWARE"]["device"]
        self.seed       = int(self.config["HARDWARE"]["seed"])
        self.n_GPUs     = int(self.config["HARDWARE"]["n_GPUs"])
        self.precision  = self.config["HARDWARE"]["precision"]

    def parseDataset(self):
        self.train_set_paths    = self.config["DATASET"]["train_set_paths"].split(',')
        self.test_set_paths     = self.config["DATASET"]["test_set_paths"].split(',')
        self.rgb_range          = int(self.config["DATASET"]["rgb_range"])
        self.batch_size         = int(self.config["DATASET"]["batch_size"])
        self.scale              = int(self.config["DATASET"]["scale"])
        self.include_noise      = self.config["DATASET"].getboolean("include_noise")
        self.noise_sigma        = float(self.config["DATASET"]["noise_sigma"])
        self.noise_mean         = float(self.config["DATASET"]["noise_mean"])
        self.include_blur       = self.config["DATASET"].getboolean("include_blur")
        self.blur_radius        = float(self.config["DATASET"]["blur_radius"])
        self.normalize          = self.config["DATASET"]["normalize"]
        self.random_flips       = self.config["DATASET"].getboolean("random_flips")
        self.channel_number     = int(self.config["DATASET"]["channel_number"])
        self.n_colors           = int(self.config["DATASET"]["n_colors"])
        self.hr_shape           = list(map(int, self.config["DATASET"]["hr_shape"].split(',')))
        self.downgrade          = self.config["DATASET"]["downgrade"]
        self.validation_size    = float(self.config["DATASET"]["validation_size"])
        self.shuffle_dataset    = self.config["DATASET"].getboolean("shuffle_dataset")

    def parseModel(self):
        self.model          = self.config["MODEL"]["model"]
        self.self_ensemble  = self.config["MODEL"].getboolean("self_ensemble")
        if self.model == 'EDSR':
            self.n_resblocks    = int(self.config["EDSR"]["n_resblocks"])
            self.n_feats        = int(self.config["EDSR"]["n_feats"])
            self.res_scale      = float(self.config["EDSR"]["res_scale"])
        elif self.model == 'PFF':
            self.emb_dimension  = int(self.config["PFF"]["emb_dimension"])
            self.filterSize     = int(self.config["PFF"]["filterSize"])
            self.pretrained     = self.config["PFF"].getboolean("pretrained")
        else:
            self.n_resblocks    = 32
            self.n_feats        = 256
            self.res_scale      = 0.1
            self.emb_dimension  = 16
            self.filterSize     = 17
            self.pretrained     = True

    def parseOptimization(self):
        self.learning_rate          = float(self.config["OPTIMIZATION"]["learning_rate"])
        self.decay                  = self.config["OPTIMIZATION"]["decay"]
        self.decay_factor_gamma     = float(self.config["OPTIMIZATION"]["decay_factor_gamma"])
        self.optimizer              = self.config["OPTIMIZATION"]["optimizer"]
        self.momentum               = float(self.config["OPTIMIZATION"]["momentum"])
        self.betas                  = list(map(float, self.config["OPTIMIZATION"]["betas"].split(',')))
        self.epsilon                = float(self.config["OPTIMIZATION"]["epsilon"])
        self.weight_decay           = float(self.config["OPTIMIZATION"]["weight_decay"])
        self.gclip                  = float(self.config["OPTIMIZATION"]["gclip"])

    def parseCheckpoint(self):
        if self.config["CHECKPOINT"].getboolean("load"):
            self.load = "./ckpts/" + self.model + "x{}".format(self.scale) + "/" + self.experiment_name
        else:
            self.load = ''
        if self.config["CHECKPOINT"].getboolean("save"):
            self.save = "./ckpts/" + self.model + "x{}".format(self.scale) + "/" + self.experiment_name
        else:
            self.save = ''

        self.reset      = self.config["CHECKPOINT"].getboolean("reset")
        self.data_test  = self.config["CHECKPOINT"]["data_test"]

    def parseTraining(self):
        self.training           = self.config["TRAINING"].getboolean("training")
        self.epoch_num          = int(self.config["TRAINING"]["epoch_num"])
        self.loss               = self.config["TRAINING"]["loss"]
        self.skip_thr           = float(self.config["TRAINING"]["skip_thr"])
        self.image_range        = int(self.config["TRAINING"]["image_range"])
        self.log_every          = int(self.config["TRAINING"]["log_every"])
        self.validate_every     = float(self.config["TRAINING"]["validate_every"])
        self.log_psnr           = self.config["TRAINING"].getboolean("log_psnr")
        self.log_ssim           = self.config["TRAINING"].getboolean("log_ssim")
        self.save_path          = "/experiment/" + self.experiment_name
        self.pre_train          = self.config["TRAINING"]["pre_train"]
        self.only_body          = self.config["TRAINING"].getboolean("only_body")
        self.fine_tuning        = self.config["TRAINING"].getboolean("fine_tuning")
        self.freeze_initial_layers = self.config["TRAINING"].getboolean("freeze_initial_layers")
        self.chop               = self.config["TRAINING"].getboolean("chop")
        self.save_models        = self.config["TRAINING"].getboolean("save_models")

        if self.pre_train in ["load_latest", "load_best"]:
            self.resume = -1  # -1 --> load latest, 0--> load pre-trained model , else --> load from desired epoch
        else:
            self.resume = 0

    def parseTesting(self):
        self.test_only              = self.config["TESTING"].getboolean("test_only")
        self.log_test_result        = self.config["TESTING"].getboolean("log_test_result")
        self.test_single            = self.config["TESTING"].getboolean("test_single")
        self.test_psnr              = self.config["TESTING"].getboolean("test_psnr")
        self.test_ssim              = self.config["TESTING"].getboolean("test_ssim")
        self.test_visualize         = self.config["TESTING"].getboolean("test_visualize")
        self.test_image_save        = self.config["TESTING"].getboolean("test_image_save")
